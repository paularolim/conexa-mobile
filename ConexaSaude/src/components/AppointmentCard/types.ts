import { TouchableOpacityProps } from 'react-native';

export interface AppointmentCardProps extends TouchableOpacityProps {
  date: string;
  patient: string;
}
