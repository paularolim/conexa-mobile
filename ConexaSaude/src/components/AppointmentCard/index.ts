export * from './AppointmentCard';
export * from './types';
