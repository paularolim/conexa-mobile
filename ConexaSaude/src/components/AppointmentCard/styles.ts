import { TouchableOpacity, View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(TouchableOpacity)`
  background-color: ${({ theme }) => theme.colors.white};
  width: 100%;
  padding: 16px;
  gap: 12px;
  border-radius: 12px;
  border: 0.5px solid ${({ theme }) => theme.colors.grayLight};
`;

export const Section = styled(View)`
  flex-direction: row;
  align-items: center;
  gap: 8px;
`;

export const DateTitle = styled(Text)`
  color: ${({ theme }) => theme.colors.gray};
  font-size: 12px;
  font-weight: bold;
`;

export const Avatar = styled(View)`
  background-color: ${({ theme }) => theme.colors.grayLight};
  width: 40px;
  height: 40px;
  border-radius: 20px;
  overflow: hidden;
`;
