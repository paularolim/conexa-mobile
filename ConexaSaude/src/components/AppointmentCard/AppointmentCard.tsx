import { View } from 'react-native';
import { useTheme } from 'styled-components/native';

import { Divider } from '@components/Divider/Divider';
import { Icon } from '@components/Icon';
import { Text } from '@components/Text';

import { Avatar, Container, DateTitle, Section } from './styles';
import { AppointmentCardProps } from './types';

export function AppointmentCard({ date, patient, ...rest }: AppointmentCardProps) {
  const { colors } = useTheme();

  return (
    <Container {...rest}>
      <Section>
        <Icon name="calendar" width={28} height={28} fill={colors.grayLight} />
        <View>
          <DateTitle>Data</DateTitle>
          <Text>{date}</Text>
        </View>
      </Section>
      <Divider />
      <Section>
        <Avatar>
          <Icon name="user" fill={colors.pink} />
        </Avatar>
        <Text>{patient}</Text>
      </Section>
    </Container>
  );
}
