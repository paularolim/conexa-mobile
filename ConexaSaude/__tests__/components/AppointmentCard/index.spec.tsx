import renderer from 'react-test-renderer';

import { AppointmentCard } from '@components/AppointmentCard';

import { Wrapper } from '../../Wrapper';

describe('AppointmentCard component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <AppointmentCard date="2023-10-01 8:30" patient="Mocked Name" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
