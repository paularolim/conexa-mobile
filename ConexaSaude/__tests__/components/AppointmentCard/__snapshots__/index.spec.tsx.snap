// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentCard component should render default component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#C2CFE0",
      "borderRadius": 12,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "gap": 12,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        },
      ]
    }
  >
    <svg
      data-file-name="SvgCalendar"
      height={28}
      stroke="#C2CFE0"
      width={28}
    />
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#707683",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Data
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
          ]
        }
      >
        2023-10-01 8:30
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "gap": 10,
        },
      ]
    }
  >
    <View
      primary={true}
      style={
        [
          {
            "backgroundColor": "#F28080",
            "borderRadius": 2,
            "height": 5,
            "width": 80,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#C2CFE0",
            "borderRadius": 2,
            "height": 5,
            "width": 80,
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#C2CFE0",
            "borderRadius": 20,
            "height": 40,
            "overflow": "hidden",
            "width": 40,
          },
        ]
      }
    >
      <svg
        data-file-name="SvgUser"
        height={40}
        stroke="#F28080"
        width={40}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#000000",
          },
        ]
      }
    >
      Mocked Name
    </Text>
  </View>
</View>
`;
